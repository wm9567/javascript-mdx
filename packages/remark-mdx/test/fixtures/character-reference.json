{
  "type": "root",
  "children": [
    {
      "type": "paragraph",
      "children": [
        {
          "type": "text",
          "value": "The characters ",
          "position": {
            "start": {
              "line": 1,
              "column": 1,
              "offset": 0
            },
            "end": {
              "line": 1,
              "column": 16,
              "offset": 15
            },
            "indent": []
          }
        },
        {
          "type": "inlineCode",
          "value": ">",
          "position": {
            "start": {
              "line": 1,
              "column": 16,
              "offset": 15
            },
            "end": {
              "line": 1,
              "column": 19,
              "offset": 18
            },
            "indent": []
          }
        },
        {
          "type": "text",
          "value": " and ",
          "position": {
            "start": {
              "line": 1,
              "column": 19,
              "offset": 18
            },
            "end": {
              "line": 1,
              "column": 24,
              "offset": 23
            },
            "indent": []
          }
        },
        {
          "type": "inlineCode",
          "value": "}",
          "position": {
            "start": {
              "line": 1,
              "column": 24,
              "offset": 23
            },
            "end": {
              "line": 1,
              "column": 27,
              "offset": 26
            },
            "indent": []
          }
        },
        {
          "type": "text",
          "value": " don’t need to be handled differently: >, }.",
          "position": {
            "start": {
              "line": 1,
              "column": 27,
              "offset": 26
            },
            "end": {
              "line": 1,
              "column": 71,
              "offset": 70
            },
            "indent": []
          }
        }
      ],
      "position": {
        "start": {
          "line": 1,
          "column": 1,
          "offset": 0
        },
        "end": {
          "line": 1,
          "column": 71,
          "offset": 70
        },
        "indent": []
      }
    },
    {
      "type": "blockquote",
      "children": [
        {
          "type": "paragraph",
          "children": [
            {
              "type": "text",
              "value": "In fact, ",
              "position": {
                "start": {
                  "line": 3,
                  "column": 3,
                  "offset": 74
                },
                "end": {
                  "line": 3,
                  "column": 12,
                  "offset": 83
                },
                "indent": []
              }
            },
            {
              "type": "inlineCode",
              "value": ">",
              "position": {
                "start": {
                  "line": 3,
                  "column": 12,
                  "offset": 83
                },
                "end": {
                  "line": 3,
                  "column": 15,
                  "offset": 86
                },
                "indent": []
              }
            },
            {
              "type": "text",
              "value": " can start a block quote!",
              "position": {
                "start": {
                  "line": 3,
                  "column": 15,
                  "offset": 86
                },
                "end": {
                  "line": 3,
                  "column": 40,
                  "offset": 111
                },
                "indent": []
              }
            }
          ],
          "position": {
            "start": {
              "line": 3,
              "column": 3,
              "offset": 74
            },
            "end": {
              "line": 3,
              "column": 40,
              "offset": 111
            },
            "indent": []
          }
        }
      ],
      "position": {
        "start": {
          "line": 3,
          "column": 1,
          "offset": 72
        },
        "end": {
          "line": 3,
          "column": 40,
          "offset": 111
        },
        "indent": []
      }
    },
    {
      "type": "paragraph",
      "children": [
        {
          "type": "text",
          "value": "Those characters are also fine in JSX elements: ",
          "position": {
            "start": {
              "line": 5,
              "column": 1,
              "offset": 113
            },
            "end": {
              "line": 5,
              "column": 49,
              "offset": 161
            },
            "indent": []
          }
        },
        {
          "type": "mdxSpanElement",
          "name": null,
          "attributes": [],
          "children": [
            {
              "type": "text",
              "value": "> and }",
              "position": {
                "start": {
                  "line": 5,
                  "column": 51,
                  "offset": 163
                },
                "end": {
                  "line": 5,
                  "column": 58,
                  "offset": 170
                },
                "indent": []
              }
            }
          ],
          "position": {
            "start": {
              "line": 5,
              "column": 49,
              "offset": 161
            },
            "end": {
              "line": 5,
              "column": 61,
              "offset": 173
            },
            "indent": []
          }
        },
        {
          "type": "text",
          "value": ".",
          "position": {
            "start": {
              "line": 5,
              "column": 61,
              "offset": 173
            },
            "end": {
              "line": 5,
              "column": 62,
              "offset": 174
            },
            "indent": []
          }
        }
      ],
      "position": {
        "start": {
          "line": 5,
          "column": 1,
          "offset": 113
        },
        "end": {
          "line": 5,
          "column": 62,
          "offset": 174
        },
        "indent": []
      }
    },
    {
      "type": "mdxBlockExpression",
      "value": "/* starts an expression, it must be closed: */",
      "position": {
        "start": {
          "line": 7,
          "column": 1,
          "offset": 176
        },
        "end": {
          "line": 7,
          "column": 49,
          "offset": 224
        },
        "indent": []
      }
    },
    {
      "type": "paragraph",
      "children": [
        {
          "type": "text",
          "value": "If you don’t want an expression, encode it, like so: ",
          "position": {
            "start": {
              "line": 9,
              "column": 1,
              "offset": 226
            },
            "end": {
              "line": 9,
              "column": 54,
              "offset": 279
            },
            "indent": []
          }
        },
        {
          "type": "text",
          "value": "{",
          "position": {
            "start": {
              "line": 9,
              "column": 54,
              "offset": 279
            },
            "end": {
              "line": 9,
              "column": 60,
              "offset": 285
            },
            "indent": []
          }
        },
        {
          "type": "text",
          "value": ".",
          "position": {
            "start": {
              "line": 9,
              "column": 60,
              "offset": 285
            },
            "end": {
              "line": 9,
              "column": 61,
              "offset": 286
            },
            "indent": []
          }
        }
      ],
      "position": {
        "start": {
          "line": 9,
          "column": 1,
          "offset": 226
        },
        "end": {
          "line": 9,
          "column": 61,
          "offset": 286
        },
        "indent": []
      }
    },
    {
      "type": "mdxBlockElement",
      "name": "starts",
      "attributes": [
        {
          "type": "mdxAttribute",
          "name": "a",
          "value": "tag, which must be properly closed: ",
          "position": {
            "start": {
              "line": 11,
              "column": 9,
              "offset": 296
            },
            "end": {
              "line": 11,
              "column": 49,
              "offset": 336
            }
          }
        }
      ],
      "children": [],
      "position": {
        "start": {
          "line": 11,
          "column": 1,
          "offset": 288
        },
        "end": {
          "line": 11,
          "column": 51,
          "offset": 338
        },
        "indent": []
      }
    },
    {
      "type": "paragraph",
      "children": [
        {
          "type": "text",
          "value": "If you don’t want a tag, encode it, like so: ",
          "position": {
            "start": {
              "line": 13,
              "column": 1,
              "offset": 340
            },
            "end": {
              "line": 13,
              "column": 46,
              "offset": 385
            },
            "indent": []
          }
        },
        {
          "type": "text",
          "value": "<",
          "position": {
            "start": {
              "line": 13,
              "column": 46,
              "offset": 385
            },
            "end": {
              "line": 13,
              "column": 50,
              "offset": 389
            },
            "indent": []
          }
        },
        {
          "type": "text",
          "value": ".",
          "position": {
            "start": {
              "line": 13,
              "column": 50,
              "offset": 389
            },
            "end": {
              "line": 13,
              "column": 51,
              "offset": 390
            },
            "indent": []
          }
        }
      ],
      "position": {
        "start": {
          "line": 13,
          "column": 1,
          "offset": 340
        },
        "end": {
          "line": 13,
          "column": 51,
          "offset": 390
        },
        "indent": []
      }
    },
    {
      "type": "paragraph",
      "children": [
        {
          "type": "text",
          "value": "Those characters are fine in code: ",
          "position": {
            "start": {
              "line": 15,
              "column": 1,
              "offset": 392
            },
            "end": {
              "line": 15,
              "column": 36,
              "offset": 427
            },
            "indent": []
          }
        },
        {
          "type": "inlineCode",
          "value": "< and {",
          "position": {
            "start": {
              "line": 15,
              "column": 36,
              "offset": 427
            },
            "end": {
              "line": 15,
              "column": 45,
              "offset": 436
            },
            "indent": []
          }
        },
        {
          "type": "text",
          "value": ".\nThey’re fine in JSX elements, in code, too: ",
          "position": {
            "start": {
              "line": 15,
              "column": 45,
              "offset": 436
            },
            "end": {
              "line": 16,
              "column": 45,
              "offset": 482
            },
            "indent": [
              1
            ]
          }
        },
        {
          "type": "mdxSpanElement",
          "name": null,
          "attributes": [],
          "children": [
            {
              "type": "inlineCode",
              "value": "< and {",
              "position": {
                "start": {
                  "line": 16,
                  "column": 47,
                  "offset": 484
                },
                "end": {
                  "line": 16,
                  "column": 56,
                  "offset": 493
                },
                "indent": []
              }
            }
          ],
          "position": {
            "start": {
              "line": 16,
              "column": 45,
              "offset": 482
            },
            "end": {
              "line": 16,
              "column": 59,
              "offset": 496
            },
            "indent": []
          }
        },
        {
          "type": "text",
          "value": ".\nOr in block JSX:",
          "position": {
            "start": {
              "line": 16,
              "column": 59,
              "offset": 496
            },
            "end": {
              "line": 17,
              "column": 17,
              "offset": 514
            },
            "indent": [
              1
            ]
          }
        }
      ],
      "position": {
        "start": {
          "line": 15,
          "column": 1,
          "offset": 392
        },
        "end": {
          "line": 17,
          "column": 17,
          "offset": 514
        },
        "indent": [
          1,
          1
        ]
      }
    },
    {
      "type": "mdxBlockElement",
      "name": "B",
      "attributes": [],
      "children": [
        {
          "type": "code",
          "lang": null,
          "meta": null,
          "value": "< and {",
          "position": {
            "start": {
              "line": 20,
              "column": 1,
              "offset": 520
            },
            "end": {
              "line": 22,
              "column": 6,
              "offset": 541
            },
            "indent": [
              1,
              1
            ]
          }
        },
        {
          "type": "code",
          "lang": "tildes",
          "meta": null,
          "value": "< and {",
          "position": {
            "start": {
              "line": 24,
              "column": 1,
              "offset": 543
            },
            "end": {
              "line": 26,
              "column": 6,
              "offset": 570
            },
            "indent": [
              1,
              1
            ]
          }
        },
        {
          "type": "paragraph",
          "children": [
            {
              "type": "text",
              "value": "Inline ",
              "position": {
                "start": {
                  "line": 28,
                  "column": 1,
                  "offset": 572
                },
                "end": {
                  "line": 28,
                  "column": 10,
                  "offset": 581
                },
                "indent": []
              }
            },
            {
              "type": "inlineCode",
              "value": "code: < and {",
              "position": {
                "start": {
                  "line": 28,
                  "column": 10,
                  "offset": 581
                },
                "end": {
                  "line": 28,
                  "column": 25,
                  "offset": 596
                },
                "indent": []
              }
            },
            {
              "type": "text",
              "value": ".",
              "position": {
                "start": {
                  "line": 28,
                  "column": 25,
                  "offset": 596
                },
                "end": {
                  "line": 28,
                  "column": 26,
                  "offset": 597
                },
                "indent": []
              }
            }
          ],
          "position": {
            "start": {
              "line": 28,
              "column": 1,
              "offset": 572
            },
            "end": {
              "line": 28,
              "column": 26,
              "offset": 597
            },
            "indent": []
          }
        }
      ],
      "position": {
        "start": {
          "line": 19,
          "column": 1,
          "offset": 516
        },
        "end": {
          "line": 29,
          "column": 5,
          "offset": 602
        },
        "indent": [
          1,
          1,
          1,
          1,
          1,
          1,
          1,
          1,
          1,
          1
        ]
      }
    }
  ],
  "position": {
    "start": {
      "line": 1,
      "column": 1,
      "offset": 0
    },
    "end": {
      "line": 30,
      "column": 1,
      "offset": 603
    }
  }
}
