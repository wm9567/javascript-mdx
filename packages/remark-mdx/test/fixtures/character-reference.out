The characters `>` and `}` don’t need to be handled differently: >, }.

> In fact, `>` can start a block quote!

Those characters are also fine in JSX elements: <>> and }</>.

{/* starts an expression, it must be closed: */}

If you don’t want an expression, encode it, like so: \{.

<starts a="tag, which must be properly closed: "/>

If you don’t want a tag, encode it, like so: \<.

Those characters are fine in code: `< and {`.
They’re fine in JSX elements, in code, too: <>`< and {`</>.
Or in block JSX:

<B>
  ```
  < and {
  ```

  ```tildes
  < and {
  ```

  Inline `code: < and {`.
</B>
